contactsummary/js/test/controllers/key-dates.controller.spec.js

```diff
@@ -31,13 +30,30 @@ define([
     }));
 
     describe('constructor', function () {
-      it('Should subscribe for contract changes', function () {
+      beforeEach(function () {
         spyOn(contractServiceMock, 'get').and.callThrough();
-        controllerObj = ctrlConstructor('KeyDatesController');
-        expect(PubSubMock.subscribe).toHaveBeenCalledWith('contract-refresh', jasmine.any(Function));
-        expect(controllerObj.dates).toEqual([]);
+        controller = ctrlConstructor('KeyDatesController');
+      });
+
+      it('Should subscribe for contract changes', function () {
+        expect(controller.dates).toEqual([]);
         expect(contractServiceMock.get).toHaveBeenCalled();
       });
+
+      it('initialiaze contracts and job roles count and key dates', function () {
+        expect(controller.activeContracts).toBe(0);
+        expect(controller.activeRoles).toBe(0);
+        expect(controller.dates.length).toBe(0);
+      });
+
+      it('initialiaze pubsub subscriptions', function () {
+        expect(PubSubMock.subscribe).toHaveBeenCalledWith('Contract::created', jasmine.any(Function));
+        expect(PubSubMock.subscribe).toHaveBeenCalledWith('Contract::updated', jasmine.any(Function));
+        expect(PubSubMock.subscribe).toHaveBeenCalledWith('Contract::deleted', jasmine.any(Function));
+        expect(PubSubMock.subscribe).toHaveBeenCalledWith('JobRole::created', jasmine.any(Function));
+        expect(PubSubMock.subscribe).toHaveBeenCalledWith('JobRole::updated', jasmine.any(Function));
+        expect(PubSubMock.subscribe).toHaveBeenCalledWith('JobRole::deleted', jasmine.any(Function));
+      })
```

Comment:
missing `;`

Reply:
**Important:** we are missing tests for what happens when these events are fired. You need at least two:

```js
describe('when contracts or job roles are updated', function () {
  beforeEach(function () {
    var eventNames = ['Contract::created', 'JobRole::created', /* ..... */];

    eventNames.forEach(function (eventName) {
      pubSubMock.publish(eventName); 
    }
  });

  it('.... what does it do when they are updated?');
});
```

Same thing for the delete scenario.

Sidenote: I'd use the real PubSub service here.

Reply:
Added test cases!